frontera/contrib/backends/memory/__init__.py

```diff
@@ -1,14 +1,21 @@
 from __future__ import absolute_import
 import logging
 import random
-from collections import deque
+from collections import deque, Iterable
 
 from frontera.contrib.backends import CommonBackend
 from frontera.core.components import Metadata, Queue, States
 from frontera.core import OverusedBuffer
 from frontera.utils.heap import Heap
 from frontera.contrib.backends.partitioners import Crc32NamePartitioner
 from frontera.utils.url import parse_domain_from_url_fast
+import six
+from six.moves import map
+from six.moves import range
+
+
+def cmp(a, b):
```

Comment:
In python 3 cmp is not an in-built function. It is used here - https://github.com/Preetwinder/frontera/blob/ec0811a21983ab724948ccb7a9e8fdea759e42b9/frontera/contrib/backends/memory/__init__.py#L80 and a few other locations below this line.